core/test/dataset_test_common.cpp

```diff
@@ -6,67 +6,97 @@
+  base.setMask("masks_z", createVariable<bool>(Dimensions{Dim::Z, lz},
+                                               Values(bZ.begin(), bZ.end())));
+
+  vX = rand(lx);
+  base.setAttr("attr_scalar", createVariable<double>(Values{rand(1).front()}));
+  base.setAttr("attr_x", createVariable<double>(Dimensions{Dim::X, lx},
+                                                Values(vX.begin(), vX.end())));
 }
 
 Dataset DatasetFactory3D::make() {
   Dataset dataset(base);
+  auto vX = rand(lx);
+  auto dvalX = rand(lx);
+  auto dvarX = rand(lx);
+  auto dvalXY = rand(lx * ly);
+  auto dvarXY = rand(lx * ly);
+  auto dvalXYZ = rand(lx * ly * lz);
+  auto dvalZYX = rand(lx * ly * lz);
+  auto dvarZYX = rand(lx * ly * lz);
   dataset.setData("values_x",
-                  makeVariable<double>({Dim::X, lx}, rand(lx)) /*LABEL_1*/);
-  dataset.setData("data_x", makeVariable<double>({Dim::X, lx}, rand(lx),
-                                                 rand(lx)) /*LABEL_1*/);
+                  createVariable<double>(Dimensions{Dim::X, lx},
+                                         Values(vX.begin(), vX.end())));
+  dataset.setData(
+      "data_x", createVariable<double>(Dimensions{Dim::X, lx},
+                                       Values(dvalX.begin(), dvalX.end()),
+                                       Variances(dvarX.begin(), dvarX.end())));
 
-  dataset.setData("data_xy", makeVariable<double>({{Dim::X, lx}, {Dim::Y, ly}},
-                                                  rand(lx * ly),
-                                                  rand(lx * ly)) /*LABEL_1*/);
+  dataset.setData("data_xy", createVariable<double>(
+                                 Dimensions{{Dim::X, lx}, {Dim::Y, ly}},
+                                 Values(dvalXY.begin(), dvalXY.end()),
+                                 Variances(dvarXY.begin(), dvarXY.end())));
 
-  dataset.setData(
-      "data_zyx",
-      makeVariable<double>({{Dim::Z, lz}, {Dim::Y, ly}, {Dim::X, lx}},
-                           rand(lx * ly * lz), rand(lx * ly * lz)) /*LABEL_1*/);
+  dataset.setData("data_zyx",
+                  createVariable<double>(
+                      Dimensions{{Dim::Z, lz}, {Dim::Y, ly}, {Dim::X, lx}},
+                      Values(dvalZYX.begin(), dvalZYX.end()),
+                      Variances(dvarZYX.begin(), dvarZYX.end())));
 
-  dataset.setData("data_xyz", makeVariable<double>(
-                                  {{Dim::X, lx}, {Dim::Y, ly}, {Dim::Z, lz}},
-                                  rand(lx * ly * lz)) /*LABEL_1*/);
+  dataset.setData("data_xyz",
+                  createVariable<double>(
+                      Dimensions{{Dim::X, lx}, {Dim::Y, ly}, {Dim::Z, lz}},
+                      Values(dvalXYZ.begin(), dvalXYZ.end())));
 
-  dataset.setData("data_scalar", makeVariable<double>(rand(1).front()));
+  dataset.setData("data_scalar",
+                  createVariable<double>(Values{rand(1).front()}));
```

Comment:
The changes for `std::vector` using `begin` and `end` are really hurting readability sometimes. I just tried locally, and `createVariable` works perfectly fine with something like  `Values(std::vector<int>{1,2,3})`, etc. Why are you not using it?

Reply:
Well, at some point it doesn't work. I'm not sure I understand why it is working, but if it is I would fix that in separate issue.

Reply:
I would not want to merge massive changes if we are going to undo them right away. Why does it work for me but not for you?

Reply:
What is the exact code working for `Values(std::vector<int>{1,2,3})` for what revision. This:
```
auto var = createVariable<float>(Values(std::vector<float>{1, 2, 3}));
```
throws.

Reply:
Missing `Dims` and `Shape`?

Reply:
this:
```
 auto var = createVariable<float>(Dims{Dim::X}, Shape{3}, Values(std::vector<float>{1, 2, 3}));
```
also throws.

Reply:
This works for me:
```cpp
TEST(VariableUniversalConstructorTest, from_vector) {
  EXPECT_EQ(createVariable<double>(Dims{Dim::X}, Shape{3},
                                   Values(std::vector<int>{1, 2, 3})),
            createVariable<double>(Dims{Dim::X}, Shape{3}, Values({1, 2, 3})));
}
```